### PR TITLE
Backup Modal Variant Testing

### PR DESCRIPTION
**Pull request by Jon-edge**
image
image
image
image

### CHANGELOG

Does this branch warrant an entry to the CHANGELOG?

- [x] Yes
- [ ] No

### Dependencies
https://github.com/EdgeApp/edge-react-gui/pull/4960

### Requirements

If you have made **any** visual changes to the GUI. Make sure you have:

- [x] Tested on iOS device
- [ ] Tested on Android device
- [ ] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206548591329441

### CODE REVIEW

File: src/components/modals/ButtonsModal.tsx [hunk 1]
```diff
@@ -23,14 +23,19 @@ export interface ButtonInfo {
 
 export interface ButtonModalProps<Buttons> {
   bridge: AirshipBridge<keyof Buttons | undefined>
-  title?: string
-  message?: string
-  children?: React.ReactNode
   buttons: Buttons
+  /** Ideally mutually exclusive with message/subtext since children are
+   * rendered in the body of the modal along with message/subtext */
```
Comment (swansontec) [inline src/components/modals/ButtonsModal.tsx:28]: /** Used to pass non-text children, to be rendered after the title and message but before the buttons themselves. */

File: src/components/modals/ButtonsModal.tsx
Comment (swansontec) [inline src/components/modals/ButtonsModal.tsx:1]: A lot of these changes seem out-of-scope for the rename. I think this `ButtonsModal` cleanup needs to be in its own commit.

File: src/actions/ScamWarningActions.tsx [hunk 2]
```diff
@@ -23,7 +23,7 @@ export const triggerScamWarningModal = async (disklet: Disklet) => {
 
       return (
         <ConfirmContinueModal bridge={bridge} title={lstrings.warning_scam_title}>
-          <Paragraph isWarning>{warningMessage}</Paragraph>
+          <Paragraph fontType="warning">{warningMessage}</Paragraph>
```
Comment (swansontec) [inline src/actions/ScamWarningActions.tsx:26]: The `Paragraph` component doesn't take a `fontType` prop yet. Did this get into the wrong commit?

File: src/components/themed/EdgeText.tsx [hunk 3]
```diff
@@ -4,6 +4,8 @@ import { Platform, StyleProp, Text, TextProps, TextStyle } from 'react-native'
 import { fixSides, mapSides, sidesToPadding } from '../../util/sides'
 import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
 
+export type ParagraphFontType = 'default' | 'warning' | 'finePrint'
```
Comment (swansontec) [inline src/components/themed/EdgeText.tsx:7]: Can't we do this the same way the Web platform handles it? Define components like `<BoldText>`, `<WarningText>`, `<FinePrint>`, which can be used inline. Then it's:

```html
<Paragraph>
  <WarningText>
    Do not lose your keys, or you will <BoldText>not</BoldText> be able to get them back!
  </WarningText>
</Paragraph>
```

Your `ParagraphFontType` way might be OK too, especially if we really lock down what combinations are valid or not, but it could also end up growing out of control. It's hard to predict how these things play out.

File: src/components/themed/EdgeText.tsx [hunk 4]
```diff
@@ -45,7 +45,7 @@ export const Paragraph = (props: {
    * Specific font style definitions for size and color:
    * - default: Default EdgeText style
    * - warning: Orange font color
-   * - finePrint: Smaller, dimmed font
+   * - finePrint: Smaller, orange font
```
Comment (swansontec) [inline src/components/themed/EdgeText.tsx:48]: If it's orange, should it be called "warning" instead? Or even `<FinePrint><WarningText>{message}</WarningText></FinePrint>`?

Sometime the fine print is a warning, and sometimes it isn't, so I personally like the mix & match approach.

**Review by swansontec — CHANGES_REQUESTED** [review]
Thanks for finally tackling out text situation. I strongly support the new `Paragraph` component, but there are one or two things we should fix up.

File: src/components/themed/EdgeText.tsx (hunk 3 above)
Comment (Jon-edge) [reply src/components/themed/EdgeText.tsx:7]: Yeah I was struggling trying to define the rules and wasn't aware the suggested was possible. Works way better, thanks

File: src/components/modals/ButtonsModal.tsx
Comment (Jon-edge) [reply src/components/modals/ButtonsModal.tsx:1]: Fair point, it just kept ballooning out as I was working on this.

**Comment by Jon-edge** [comment]
I created a new PR to separate the changes not specific to this backup modal task: https://github.com/EdgeApp/edge-react-gui/pull/4960

Made this backup modal PR dependent on the above.

**Review by swansontec — APPROVED** [review]
